pydra/engine/tests/test_specs.py

```diff
@@ -172,3 +172,97 @@ def test_file_hash(tmpdir):
         inputs(in_file=outfile).hash
         == "088625131e6718a00170ad445a9c295244dffd4e5d847c8ee4b1606d623dacb1"
     )
+
+
+def test_input_file_hash_2(tmpdir):
+    """ input spec with File types, checking when the checksum changes"""
+    file = tmpdir.join("in_file_1.txt")
+    with open(file, "w") as f:
+        f.write("hello")
+
+    input_spec = SpecInfo(name="Inputs", fields=[("in_file", File)], bases=(BaseSpec,))
+    inputs = make_klass(input_spec)
+
+    # checking specific hash value
+    hash1 = inputs(in_file=file).hash
+    assert hash1 == "5d2870a7376150274eac72115fbf211792a8e5f250f220b3cc11bfc1851e4b53"
+
+    # checking if different name doesn't affect the hash
+    file_diffname = tmpdir.join("in_file_2.txt")
+    with open(file_diffname, "w") as f:
+        f.write("hello")
+    hash2 = inputs(in_file=file_diffname).hash
+    assert hash1 == hash2
+
+    # checking if different content (the same name) affects the hash
+    file_diffcontent = tmpdir.join("in_file_1.txt")
+    with open(file_diffcontent, "w") as f:
+        f.write("hi")
+    hash3 = inputs(in_file=file_diffcontent).hash
+    assert hash1 != hash3
+
+
+def test_input_file_hash_2a(tmpdir):
+    """ input spec with ty.Union[File, ...] type, checking when the checksum changes"""
+    file = tmpdir.join("in_file_1.txt")
+    with open(file, "w") as f:
+        f.write("hello")
+
+    input_spec = SpecInfo(
+        name="Inputs", fields=[("in_file", ty.Union[File, int])], bases=(BaseSpec,)
+    )
+    inputs = make_klass(input_spec)
+
+    # checking specific hash value
+    hash1 = inputs(in_file=file).hash
```

Comment:
perhaps also check when the input is an `int` here and something else in the next test. overall the tests at least cover some basic cases.

Reply:
I added int in the next test:  https://github.com/nipype/pydra/pull/195/files/01d0c1bc084a1d43562b7dc5432539d846730deb#diff-e0f87d47951f1d013292bae836ba2c75R235
but let me add a few more tests